ui/core/src/model/variables.ts

```diff
@@ -13,29 +13,55 @@
 
 import { Definition, JsonObject } from './definitions';
 
-/**
- * Variable definition options that are common to all variables.
- */
-export interface VariableDefinition<Options extends JsonObject = JsonObject> extends Definition<Options> {
-  display: VariableDisplayOptions;
-  selection: VariableSelectionOptions;
-  capturing_regexp?: string;
+export const DEFAULT_ALL_VALUE = '$__all' as const;
+
+export type VariableName = string;
+
+export type VariableState = {
+  value: VariableValue;
+  options?: VariableOption[];
+  loading: boolean;
+  error?: Error;
+};
+
+export type VariableOption = { label: string; value: string };
+
+export type VariablesState = Record<VariableName, VariableState>;
```

Comment:
the plural `s` in `VariablesState` is so subtle i feel like i could miss it, but trying to think of a good alternative too...

throwing it out there, `VariableStates`?

Reply:
Ah yeah I felt the same way. That could work, or maybe even `VariableStateMap` ?

Reply:
i like that too, `SomethingMap` was coming up when i was running through names